### PR TITLE
[fix](stats) Release analyze tasks once job finished

### PR DESCRIPTION
**Pull request by Kikyou1997**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
run buildall

**Comment by shuke987** [comment]
run feut

**Comment by Kikyou1997** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisJob.java [hunk 1]
```diff
@@ -187,7 +187,7 @@ public void cancel() {
     }
 
     public void deregisterJob() {
-        analysisManager.removeJob(jobInfo.jobId);
+        analysisManager.analysisJobIdToTaskMap.remove(jobInfo.jobId);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisJob.java:190]: we'd better add back removeJob method in AnalysisManager, and change the code inner AnalysisManager#removeJob

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
https://github.com/apache/doris/pull/27486 Same PR has been merge for master, closed

**Pull request closed without merging**